src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewComponent.cs

```diff
@@ -237,7 +237,7 @@ public ContentViewComponentResult Content(string content)
         /// <returns>A <see cref="ViewViewComponentResult"/>.</returns>
         public ViewViewComponentResult View()
         {
-            return View<object>(null, null);
+            return View(null, ViewData.Model);
```

Comment:
Maintain the `ViewDataDictionary` type coercion i.e. leave `<object>` and name the parameters.